### ISSUE TITLE
Add audit events for PaymentStart update

### ISSUE THREAD

**Issue by lsampras** [juspay/hyperswitch#4681]

**Comment by kaustubh1106** [comment]
Hey @lsampras , I need some clarity as I gone through the reference #4166 ,I have to make changes in Pyament_update.rs in updateTracker trait ?? am I right? and also I want to ask do I need to add enum in Audit_event or I have to use any existing enum???

**Comment by lsampras** [comment]
@kaustubh1106 yeah you are right about the trait there, and you can add another variant in the same enum...

**Comment by kaustubh1106** [comment]
@lsampras please review my PR

**Comment by kaustubh1106** [comment]
@lsampras hey! can you please review and tell me if there is any other changes I have to make?

**Comment by kaustubh1106** [comment]
hey @lsampras can you please clarify me about this https://github.com/juspay/hyperswitch/pull/4800/files/7b508ea5d4264ee576e1b0c9e58ed646b8ccc7ae#r1621028135

**Comment by kaustubh1106** [comment]
@lsampras sir can please tell me if there is something wrong with the PR?

**Comment by lsampras** [comment]
@kaustubh1106 no worries everything seems correct, I had approved the PR as well,
there's some CI failures, would you be able to take a look at that?

**Comment by kaustubh1106** [comment]
@lsampras [Check compilation on MSRV toolchain (ubuntu-latest)](https://github.com/juspay/hyperswitch/actions/runs/9361343241/job/25768165782#logs)
Sir I am unable to fix this and 
<img width="939" alt="Screenshot 2024-06-04 at 5 10 32 PM" src="https://github.com/juspay/hyperswitch/assets/123895549/1ab952a5-e4aa-4c8b-a3e9-5b56e157945c">
and following error req_state i think you suggested me

**Comment by lsampras** [comment]
@kaustubh1106 you should be able to fix it by adding the suggested import 
(`use crate: events: :audit_events: :AuditEventType;`),
can test it locally via `make clippy`

**Comment by kaustubh1106** [comment]
@lsampras sir can you please review the PR

**Comment by kaustubh1106** [comment]
@lsampras
